include/zephyr/pixel/image.h

```diff
@@ -0,0 +1,203 @@
+ *
+ * @param img Image to convert.
+ * @param window_size The window size for convnerting, usually 2 (faster) or 3 (higher quality).
+ */
+int pixel_image_debayer(struct pixel_image *img, uint32_t window_size);
+
+/**
+ * @brief Resize an image.
+ *
+ * An operation is added to change the image size. The aspect ratio is not preserved and the output
+ * image size is exactly the same as requested.
+ *
+ * @param img Image to convert.
+ * @param width The new width in pixels.
+ * @param height The new height in pixels.
+ */
+int pixel_image_resize(struct pixel_image *img, uint16_t width, uint16_t height);
+
+/**
+ * @brief Apply a kernel operation on an image.
+ *
+ * Kernel operations are working on small blocks of typically 3x3 or 5x5 pixels, repeated over the
+ * entire image to apply a desired effect on an image.
+ *
+ * @param img Image to convert.
+ * @param kernel_type The type of kernel to apply as defined in <zephyr/pixel/kernel.h>
+ * @param kernel_size The size of the kernel operaiton, usually 3 or 5.
+ */
+int pixel_image_kernel(struct pixel_image *img, uint32_t kernel_type, int kernel_size);
+
+/**
+ * @brief Print an image using higher quality TRUECOLOR terminal escape codes.
+ *
+ * @param img Image to print.
+ */
+void pixel_image_print_truecolor(struct pixel_image *img);
+
+/**
+ * @brief Print an image using higher speed 256COLOR terminal escape codes.
+ *
+ * @param img Image to print.
+ */
+void pixel_image_print_256color(struct pixel_image *img);
+
+/** @cond INTERNAL_HIDDEN */
+
+/**
+ * @brief Add a operation processing step to an image.
+ *
+ * @note This is a low-level function only needed to implement new operations.
+ *
+ * The operation step will not be processed immediately, but rather added to a linked list of
+ * operations that will be performed at runtime.
+ *
+ * @param img Image to which add a processing step.
+ * @param template Stream processing step to apply to the image.
+ * @param buffer_size Size of the input buffer to allocate for this operation.
+ * @param threshold Minimum number of bytes the operation needs to run one cycle.
+ * @return 0 on success
+ */
+int pixel_image_add_operation(struct pixel_image *img, const struct pixel_operation *template,
+			      size_t buffer_size, size_t threshold);
+
+/**
+ * @brief Add a operation processing step to an image for uncompressed input data.
+ *
+ * @note This is a low-level function only needed to implement new operations.
+ *
+ * The operation step will not be processed immediately, but rather added to a linked list of
+ * operations that will be performed at runtime.
+ *
+ * Details such as buffer size or threshold value are deduced from the stream.
+ *
+ * @param img Image to which add a processing step.
+ * @param template Stream processing step to apply to the image.
+ */
+int pixel_image_add_uncompressed(struct pixel_image *img, const struct pixel_operation *template);
+
+/**
+ * @brief Perform all the processing added to the
```

Comment:
unfinished sentence, is it triggering the processing pipeline?

Reply:
Good catch! I will check if I have more up to date documentation somewhere else and patch it.